java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java

```diff
@@ -661,7 +661,10 @@ void executeInstruction(Instruction instruction) {
       case ARRAYLENGTH:
         pop = popStack(1, instruction.opcode);
         sv = constraintManager.createSymbolicValue(instruction);
-        programState = pop.state.stackValue(sv).addConstraint(sv, ObjectConstraint.NOT_NULL);
+        programState = pop.state.stackValue(sv);
+        if(instruction.opcode != ARRAYLENGTH) {
```

Comment:
`ARRAYLENGTH` opcode has little common with the other two opcodes, so I think it will be more readable to have distinct `case` block for it instead of making this `if` inside